src/System.Console/tests/ReadAndWrite.cs

```diff
@@ -261,6 +259,30 @@ public static unsafe void ValidateConsoleEncoding(Encoding encoding)
     }
 
     [Fact]
+    // On the full framework it is not guaranteed to eat the preamble bytes
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+    public static unsafe void OutputEncodingPreamble()
+    {
+        Encoding curEncoding = Console.OutputEncoding;
+
+        try
+        {
+            Encoding encoding = Console.Out.Encoding;
+            // The primary purpose of ConsoleEncoding is to return an empty preamble.
+            Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
```

Comment:
Console.Out will create the ConsoleEncoding wrapper around Console.OutputEncoding mainly to eat the preamble. this is the behavior of netcore to ensure no preamble is produced. the desktop doesn't do that .